src/main/java/seedu/address/logic/commands/EditVolunteerCommand.java

```diff
@@ -0,0 +1,268 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_BIRTHDAY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_GENDER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VOLUNTEER_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_VOLUNTEERS;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.volunteer.Birthday;
+import seedu.address.model.volunteer.Gender;
+import seedu.address.model.volunteer.Volunteer;
+import seedu.address.model.volunteer.VolunteerAddress;
+import seedu.address.model.volunteer.VolunteerEmail;
+import seedu.address.model.volunteer.VolunteerId;
+import seedu.address.model.volunteer.VolunteerName;
+import seedu.address.model.volunteer.VolunteerPhone;
+
+/**
+ * Edits the details of an existing volunteer in the address book.
+ */
+public class EditVolunteerCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the volunteer identified "
+            + "by the index number used in the displayed volunteer list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_VOLUNTEER_NAME + "NAME] "
+            + "[" + PREFIX_VOLUNTEER_GENDER + "GENDER] "
+            + "[" + PREFIX_VOLUNTEER_BIRTHDAY + "BIRTHDAY] "
+            + "[" + PREFIX_VOLUNTEER_PHONE + "PHONE] "
+            + "[" + PREFIX_VOLUNTEER_EMAIL + "EMAIL] "
+            + "[" + PREFIX_VOLUNTEER_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_VOLUNTEER_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_VOLUNTEER_PHONE + "91234567 "
+            + PREFIX_VOLUNTEER_EMAIL + "johndoe@example.com";
+
+    public static final String MESSAGE_EDIT_VOLUNTEER_SUCCESS = "Edited Volunteer: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_VOLUNTEER = "This volunteer already exists in the address book.";
```

Comment:
Change address book to something like
"This volunteer already exists in the application"